server/src/daemon/collection.rs

```diff
@@ -0,0 +1,206 @@
+            Arc::clone(&self.keyring),
+            Arc::clone(&self.manager),
+        )
+        .await;
+        let path = OwnedObjectPath::from(item.path());
+        self.items.write().await.push(item.clone());
+        object_server.at(&path, item).await.unwrap();
+
+        Self::item_created(&ctxt, path.as_ref()).await?;
+        Ok((path, prompt))
+    }
+
+    #[zbus(property, name = "Items")]
+    pub async fn items(&self) -> Vec<OwnedObjectPath> {
+        self.items
+            .read()
+            .await
+            .iter()
+            .map(|item| OwnedObjectPath::from(item.path()))
+            .collect()
+    }
+
+    #[zbus(property, name = "Label")]
+    pub fn label(&self) -> &str {
+        &self.label
+    }
+
+    #[zbus(property, name = "Locked")]
+    pub fn locked(&self) -> bool {
+        self.locked.load(std::sync::atomic::Ordering::Relaxed)
+    }
+
+    #[zbus(property, name = "Created")]
+    pub fn created_as_secs(&self) -> u64 {
+        self.created.as_secs()
+    }
+
+    #[zbus(property, name = "Modified")]
+    pub fn modified_as_secs(&self) -> u64 {
+        self.modified.as_secs()
+    }
+
+    #[zbus(signal)]
+    pub async fn item_created(ctxt: &SignalContext<'_>, item: ObjectPath<'_>) -> zbus::Result<()>;
+
+    #[zbus(signal)]
+    pub async fn item_deleted(ctxt: &SignalContext<'_>, item: ObjectPath<'_>) -> zbus::Result<()>;
+
+    #[zbus(signal)]
+    pub async fn item_changed(ctxt: &SignalContext<'_>, item: ObjectPath<'_>) -> zbus::Result<()>;
+}
+
+impl Collection {
+    pub fn new(
+        label: &str,
+        alias: &str,
+        created: Duration,
+        keyring: Arc<Keyring>,
+        manager: Arc<Mutex<ServiceManager>>,
+    ) -> Self {
+        Self {
+            items: Default::default(),
+            label: label.to_owned(),
+            alias: Arc::new(RwLock::new(alias.to_owned())),
+            locked: Arc::new(AtomicBool::new(false)),
+            modified: created,
+            item_counter: Arc::new(RwLock::new(0)),
+            path: OwnedObjectPath::try_from(format!("{}{}", SECRET_COLLECTION_PREFIX, label))
+                .unwrap(),
+            created,
+            keyring,
+            manager,
+        }
+    }
+
+    pub fn path(&self) -> ObjectPath {
+        self.path.as_ref()
+    }
+
+    pub async fn set_alias(&self, ctxt: &zbus::SignalContext<'_>, alias: &str) -> Result<()> {
+        *self.alias.write().await = alias.to_owned();
+        Service::collection_changed(ctxt, self.path.as_ref()).await?;
+        Ok(())
+    }
+
+    pub async fn set_locked(&self, ctxt: &zbus::SignalContext<'_>, locked: bool) {
+        self.locked
+            .store(locked, std::sync::atomic::Ordering::Relaxed);
+        Service::collection_changed(ctxt, self.path.as_ref())
+            .await
+            .unwrap();
+    }
+
+    pub async fn item_counter(&self) -> i32 {
+        *self.item_counter.read().await
+    }
+
+    pub async fn set_item_counter(&self) {
```

Comment:
the method doesn't set the item counter, it increments it

Reply:
I agree. Will change it to `increment_item_counter`